observer/src/observer/daemon/daemon.py

```
@@ -0,0 +1,309 @@
+        "observer" markers. Uses /proc on Linux, falls back to `ps` on
+        macOS/other Unix. Returns False if verification cannot be performed.
+        """
+        try:
+            # Linux: read /proc/{pid}/cmdline (null-delimited)
+            cmdline_path = Path(f"/proc/{pid}/cmdline")
+            if cmdline_path.exists():
+                raw = cmdline_path.read_bytes().replace(b"\x00", b" ").decode()
+                return "observer.daemon" in raw or "observer" in raw.split()
+
+            # macOS / fallback: use ps
+            result = subprocess.run(
```

Ruff (0.15.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)